src/metric_definitions.go

```diff
@@ -5,7 +5,62 @@
 
 package main
 
-import "github.com/newrelic/infra-integrations-sdk/data/metric"
+import (
+	"reflect"
+
+	"github.com/newrelic/infra-integrations-sdk/data/metric"
+)
+
+// Definitions struct will contain the metrics that have to be collected.
+type Definitions struct {
+	Common              []Query `yaml:"common"`
+	Metrics             []Query `yaml:"metrics"`
+	ColumnFamilyMetrics []Query `yaml:"column_family_metrics"`
+}
+
+// FilterDefinitions filters the definitions of the metrics that have to be collected based on received config.
+func (d Definitions) FilterDefinitions(config Config) Definitions {
+	// Empty config, nothing to filter.
+	if reflect.DeepEqual(config, Config{}) {
```

Comment:
Would it make sense to implement `config.IsEmpty()` to just check the underlying fields and avoid using [DeepEqual](https://pkg.go.dev/reflect#DeepEqual)? It is not really important because it will just work for those particular fields, but wonder if you considered that possibility.

Reply:
what would be the concern regarding using DeepEqual in this particular scenario? 
The structure is not very complex, comes from a YAML => unlikely to cause pointer cycles and is done just once
On the other-hand with, IsEmpty, in the future, if we add more fields like a map for e.g. we will have to make sure we update that function

Reply:
I was just wondering about different approaches... Absolutely DeepEqual is easier to maintain, It could albo be problematic if it become really complex but I don't think we are extending that much. Sorry for the noise.